packages/transaction-controller/src/types.ts

```diff
@@ -689,3 +689,19 @@ export type DefaultGasEstimates = {
    */
   maxPriorityFeePerGas?: string;
 };
+
+/**
+ * Describes the event types emitted by the transaction controller.
+ */
+export enum TransactionEvent {
```

Comment:
As mentioned in the other PR, rather than using an enum for the events, we can update the type of the `hub` to limit what can be `emitted` and what listeners can be created and even pair the arguments to event names.

https://github.com/MetaMask/core/blob/1dc1e026009da51af6406f59815520e1b58b6f44/packages/transaction-controller/src/helpers/PendingTransactionTracker.ts#L44

Reply:
Thanks Matthew, added in https://github.com/MetaMask/core/pull/1894/commits/539490fa013afd326c494b867488dd612a155e57